service/rpi-sb-provisioner.sh

```diff
@@ -497,6 +509,13 @@ if [ -z "${DEMO_MODE_ONLY}" ] && [ -n "${RPI_DEVICE_FETCH_METADATA}" ]; then
     esac
 
     keywriter_log "Board is: ${BOARD_STR}, with revision number ${REVISION}. Has Processor ${PROCESSOR_STR} with Memory ${MEMORY_STR}. Was manufactured by ${MANUFACTURER_STR}"
+
+    if [ -n "${RPI_DEVICE_METADATA_CSV}" ]; then
+        if [ ! -e "${RPI_DEVICE_METADATA_CSV}" ]; then
+            echo "serial,keyhash,mac,jtag_locked,advanced_boot,boot_rom,board_attr,boardname,boardrev,processor,memory,manufacturer" > "${RPI_DEVICE_METADATA_CSV}"
```

Comment:
Do we need to consider multiple concurrent access/locking to/for this file?

rpi-sb-provisioner@.service is an instanced service unit and nothing (that I'm aware of) prevents triage.sh from running multiple instances simultaneously. The paths for our other file writes are typically parametrised with ${TARGET_DEVICE_SERIAL} so this hasn't been an issue previously.

Reply:
I swapped this to using a WAL sqlite3 database, which I believe should be resilient against this problem.